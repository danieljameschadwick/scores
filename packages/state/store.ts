import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistReducer, persistStore } from "redux-persist";
import thunk from "redux-thunk";
import themeReducer from "./reducer/ThemeReducer";

const persistConfig = {
  key: "root",
  storage: AsyncStorage,
};

const persistedReducer = persistReducer(persistConfig, themeReducer);

const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: [thunk],
});

export const persistor = persistStore(store);

export type AppState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export type AppThunk<ReturnType = void> = ThunkAction<ReturnType,
  AppState,
  unknown,
  Action<string>>;

export default store;
