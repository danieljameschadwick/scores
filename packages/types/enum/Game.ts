export enum Game {
  FOOTBALL = "FOOTBALL",
  NBA = "NBA",
  NFL = "NFL",
};
