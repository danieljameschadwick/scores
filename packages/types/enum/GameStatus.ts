export enum GAME_STATUS {
  TBD = "TBD", // Time To Be Defined
  NS = "NS", // Not Started
  "1H" = "1H", // First Half, Kick Off
  HT = "HT", // Halftime
  "2H" = "2H", // Second Half, 2nd Half Started
  ET = "ET", // Extra Time
  P = "P", // Penalty In Progress
  FT = "FT", // Match Finished
  AET = "AET", // Match Finished After Extra Time
  PEN = "PEN", // Match Finished After Penalty
  BT = "BT", // Break Time (in Extra Time)
  SUSP = "SUSP", // Match Suspended
  INT = "INT", // Match Interrupted
  PST = "PST", // Match Postponed
  CANC = "CANC", // Match Cancelled
  ABD = "ABD", // Match Abandoned
  AWD = "AWD", // Technical Loss
  WO = "WO", // WalkOver
  LIVE = "LIVE", // In Progress
};
