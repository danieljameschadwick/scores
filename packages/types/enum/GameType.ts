export enum GameType {
  FOOTBALL = "FOOTBALL",
  NBA = "NBA",
  NFL = "NFL",
};
