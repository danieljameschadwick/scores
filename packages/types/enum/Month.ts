// @TODO: temporary Month enum for queryin
export enum Month {
  AUGUST = "AUGUST",
  SEPTEMBER = "SEPTEMBER",
};
