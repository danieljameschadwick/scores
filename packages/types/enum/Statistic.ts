export enum Statistic {
  FOULS = "fouls",
  YELLOW_CARDS = "yellowCards",
  RED_CARDS = "redCards",
  OFFSIDES = "offsides",
  CORNER_KICKS = "cornerKicks",
  GOALKEEPER_SAVES = "goalkeeperSaves",
 };
