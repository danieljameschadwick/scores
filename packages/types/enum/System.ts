export enum System {
  WEB = 'web',
  iOS = 'ios',
  ANDROID = 'android',
};
