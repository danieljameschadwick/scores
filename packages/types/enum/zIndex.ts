export enum Z_INDEXES {
  BASE = 1,
  OVERLAY = 10,
  OVERLAY_COVER = 15,
  SIDE_PANEL = 20,
  MODAL = 30,
}