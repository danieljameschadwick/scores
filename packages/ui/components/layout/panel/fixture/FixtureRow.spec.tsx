import React from "react";
import { render, screen } from "@testing-library/react-native";
import { Provider } from "react-redux";
import store, { persistor } from "@scores/state/store";
import { PersistGate } from "redux-persist/integration/react";
import { GAME_RESULT } from "@scores/types/enum/GameResult";
import { FixtureRow } from "./FixtureRow";

const customRender = (team) => {
  return render(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <FixtureRow {...team} />
      </PersistGate>
    </Provider>
  );
};

describe("FixtureRow component", () => {
  it("renders base component", async () => {
    const team = {
      name: "England",
      score: 0,
      result: GAME_RESULT.DRAW,
    };
    const { queryByTestId } = customRender(team);

    expect(queryByTestId("fixture-row")).toBeTruthy();
    expect(queryByTestId("name")).toHaveTextContent("England");
    expect(queryByTestId("score")).toHaveTextContent(0);

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("renders base win component", async () => {
    const team = {
      name: "Manchester United",
      score: 3,
      result: GAME_RESULT.WIN,
    };
    const { queryByTestId } = customRender(team);

    expect(queryByTestId("fixture-row")).toBeTruthy();
    expect(queryByTestId("name")).toHaveTextContent("Manchester United");
    expect(queryByTestId("score")).toHaveTextContent(3);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
