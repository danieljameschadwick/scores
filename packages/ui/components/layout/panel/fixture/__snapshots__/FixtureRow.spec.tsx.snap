// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixtureRow component renders base component 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 2.5,
      },
    ]
  }
  testID="fixture-row"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 5,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": null,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      dataSet={
        Object {
          "media": undefined,
        }
      }
      style={
        Array [
          Array [
            Object {
              "fontSize": 16,
            },
          ],
          Object {
            "color": "#000",
          },
        ]
      }
      testID="name"
    >
      England
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "width": 25,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ],
          Object {
            "color": "#000",
          },
        ]
      }
      testID="score"
    >
      0
    </Text>
  </View>
</View>
`;

exports[`FixtureRow component renders base win component 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 2.5,
      },
    ]
  }
  testID="fixture-row"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 5,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": null,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      dataSet={
        Object {
          "media": undefined,
        }
      }
      style={
        Array [
          Array [
            Object {
              "fontSize": 16,
            },
          ],
          Object {
            "color": "#000",
          },
        ]
      }
      testID="name"
    >
      Manchester United
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "width": 25,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ],
          Object {
            "color": "#000",
          },
        ]
      }
      testID="score"
    >
      3
    </Text>
  </View>
</View>
`;
